src/test.js

```diff
@@ -205,7 +205,7 @@ Test.prototype = {
 
 			if ( hookName === "after" &&
 				hookOwner.unskippedTestsRun !== numberOfUnskippedTests( hookOwner ) - 1 &&
-				config.queue.length > 2 ) {
+				( config.queue.length > 0 || ProcessingQueue.taskCount() > 2 ) ) {
```

Comment:
Not really related to this PR since it was pre-exixting, but this took me a little while to figure out (specifically, why is `> 2` important), it _may_ warrant an inline comment...

Reply:
Agreed...should've done that last time I refactored this code